packages/text-bitmap/src/BitmapFont.ts

```diff
@@ -226,4 +236,298 @@ export class BitmapFont
+                canvas.width = BMT_DIMEN;
+                canvas.height = BMT_DIMEN;
+
+                context = canvas.getContext('2d');
+                baseTexture = new BaseTexture(canvas, { resolution });
+            }
+
+            // Measure glyph dimensions
+            const metrics = TextMetrics.measureText(chars[i], style, false, canvas);
+            const width = metrics.width;
+            const height = metrics.height;
+
+            maxCharHeight = Math.max(height + metrics.fontProperties.descent, maxCharHeight);
+
+            // Wrap line once full row has been rendered
+            if ((width * resolution) + positionX >= lineWidth)
+            {
+                --i;
+                positionY += maxCharHeight * resolution;
+                positionX = 0;
+                maxCharHeight = 0;
+                continue;
+            }
+
+            drawGlyph(// eslint-disable-line @typescript-eslint/no-use-before-define
+                canvas,
+                context,
+                metrics,
+                positionX,
+                positionY,
+                resolution,
+                style
+            );
+
+            // Unique (numeric) ID mapping to this glyph
+            const id = metrics.text.charCodeAt(0);
+
+            // Create a texture holding just the glyph
+            textures[id] = new Texture(baseTexture, new Rectangle(positionX / resolution, positionY / resolution,
+                width,
+                height));
+
+            fontData.char[id] = {
+                id,
+                page: id,
+                x: 0,
+                y: 0,
+                width,
+                height,
+                xoffset: 0,
+                yoffset: 0,
+                xadvance: width - (style.dropShadow ? style.dropShadowDistance : 0),
+            };
+            fontData.page[id] = {
+                id,
+                file: '',
+            };
+
+            positionX += (width + (2 * BMT_PADDING)) * resolution;
+        }
+
+        const font = new BitmapFont(fontData, textures);
+
+        if (name)
+        {
+            BitmapFont.available[name] = font;
+        }
+
+        return font;
+    }
 }
+
+/**
+ * Draws the glyph `metrics.text` on the given canvas.
+ *
+ * @param {HTMLCanvasElement} canvas
+ * @param {CanvasRenderingContext2D} context
+ * @param {TextMetrics} metrics
+ * @param {number} x
+ * @param {number} y
+ * @param {number} resolution
+ * @param {TextStyle} style
+ */
+function drawGlyph(
+    canvas: HTMLCanvasElement,
+    context: CanvasRenderingContext2D,
+    metrics: TextMetrics,
+    x: number,
+    y: number,
+    resolution: number,
+    style: TextStyle
+): void
+{
+    const char = metrics.text;
+    const fontProperties = metrics.fontProperties;
+
+    context.translate(x, y);
+    context.scale(resolution, resolution);
+
+    const tx = style.strokeThickness / 2;
+    const ty = -(style.strokeThickness / 2);
+
+    context.font = style.toFontString();
```

Comment:
I'm worried that we have repeated code in drawGlyph between here and 'Text'?

Reply:
I propose a future PR to refactor the shared bit better.

Reply:
Yeah, I'm ok with that. That's why in my basic implementation, I actually _do_ use PIXI.Text to render each letter, then take the texture and use it, so that you guarantee it's the same

Reply:
@themoonrat But then don't you have one texture per glyph?

Reply:
@SukantPal - I do, that's very much the downside of what I have, and why I've had an outstanding JIRA for nearly 3 years to attempt to do what you've got here! :)